spring-cloud-gcp-data-datastore/src/test/java/org/springframework/cloud/gcp/data/datastore/it/TransactionalTemplateService.java

```diff
@@ -0,0 +1,64 @@
+/*
+ *  Copyright 2018 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.data.datastore.it;
+
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cloud.gcp.data.datastore.core.DatastoreTemplate;
+import org.springframework.transaction.annotation.Transactional;
+
+import static org.junit.Assert.assertNull;
+
+/**
+ * A transactional service used for integration tests.
+ * @author Chengyuan Zhao
+ */
+public class TransactionalTemplateService {
+
+	@Autowired
+	private DatastoreTemplate datastoreTemplate;
+
+	@Transactional
+	public void testSaveAndStateConstantInTransaction(List<TestEntity> testEntities)
+			throws InterruptedException {
+
+		for (TestEntity testEntity : testEntities) {
+			assertNull(this.datastoreTemplate.findById(testEntity.getId(),
+					TestEntity.class));
+		}
+
+		this.datastoreTemplate.saveAll(testEntities);
+
+		// Because these saved entities should NOT appear when we subsequently check, we
+		// must wait a period of time that would see a non-transactional save go through.
+		Thread.sleep(DatastoreIntegrationTests.QUERY_WAIT_INTERVAL_SECONDS * 1000);
```

Comment:
That's a long period of waiting without trying. Maybe use awaitility like we do in other tests?

Reply:
I could be wrong, but the reason I didn't use await utility is that there is nothing to wait for. We're not supposed to see a condition be fulfilled, so we would end up waiting the entire period.

Reply:
I see. So we are waiting for something not to happen. How do you choose how long to wait for? Why not longer than 15 seconds and why not less? If we can't come up with a reason for the magic number, I would say it's essentially untestable.

Reply:
@meltsufin @dmitry-s PTAL. I am now waiting a period based on how long the other awaits took.

Reply:
@meltsufin PTAL, requested change made.